Source/BluetoothManager.swift

```diff
@@ -277,15 +277,15 @@ public class BluetoothManager {
      connect or has already connected.
      - returns: Observable which emits next and complete events when peripheral successfully cancelled connection.
      */
-    public func cancelConnectionToPeripheral(peripheral: Peripheral) -> Observable<Peripheral> {
+    public func cancelPeripheralConnection(_ peripheral: Peripheral) -> Observable<Peripheral> {
         let observable = Observable<Peripheral>.create { observer in
-            let disposable = self.monitorPeripheralDisconnection(peripheral).take(1).subscribe(observer)
-            self.centralManager.cancelConnection(peripheral.peripheral)
-            return AnonymousDisposable {
+            let disposable = self.monitorDisconnection(for: peripheral).take(1).subscribe(observer)
```

Comment:
Thought about that too, but was to lazy to implement it.  ;)
But it should be done.